File: qc-domain/src/main/java/com/eager/questioncloud/domain/library/repository/UserQuestionRepository.java
package com.eager.questioncloud.domain.library.repository;

import com.eager.questioncloud.domain.library.dto.UserQuestionDetail;
import com.eager.questioncloud.domain.library.model.UserQuestion;
import com.eager.questioncloud.domain.question.common.QuestionFilter;
import java.util.List;

public interface UserQuestionRepository {
    List<UserQuestion> saveAll(List<UserQuestion> userQuestionLibraries);

    Boolean isOwned(Long userId, List<Long> questionIds);

    Boolean isOwned(Long userId, Long questionId);

    List<UserQuestionDetail> getUserQuestions(QuestionFilter questionFilter);

    int countUserQuestions(QuestionFilter questionFilter);
}

File: qc-domain/qc-question/qc-question-core/src/main/kotlin/com/eager/questioncloud/question/infrastructure/repository/UserQuestionRepository.kt
package com.eager.questioncloud.question.infrastructure.repository

import com.eager.questioncloud.question.common.QuestionFilter
import com.eager.questioncloud.question.domain.UserQuestion
import com.eager.questioncloud.question.dto.UserQuestionContent

interface UserQuestionRepository {
    fun saveAll(userQuestionLibraries: List<UserQuestion>): List<UserQuestion>

    fun isOwned(userId: Long, questionIds: List<Long>): Boolean

    fun isOwned(userId: Long, questionId: Long): Boolean

    fun getUserQuestions(questionFilter: QuestionFilter): List<UserQuestionContent>

    fun countUserQuestions(questionFilter: QuestionFilter): Int

    fun deleteAllInBatch()

    fun findByQuestionIdInAndUserId(questionIds: List<Long>, userId: Long): List<UserQuestion>
}
